### PR TITLE
Fixes:

### PR DESCRIPTION
**Pull request by oleorhagen**
* Respect the wait file
* Access the deployment logger handler globally

### CODE REVIEW

File: src/mender/client/inventory.py [hunk 1]
```diff
@@ -52,5 +52,5 @@ def request(
         return None
     log.debug(f"inventory response: {r}")
     if r.status_code != 200:
-        log.error(f"Got inventory response: {r.json()}")
+        log.error(f"Got inventory response: {r.text}")
```
Comment (merlin-northern) [inline src/mender/client/inventory.py:55]: I kind of tested it and got interested ;-) this endpoint returns valid json or nothing. are you confident that it is not json that is returned?

File: src/mender/log/log.py [hunk 2]
```diff
@@ -86,7 +86,10 @@ def add_sub_updater_log(log_file):
...
             log.info(f"Sub-updater-logs follows:\n{log_string}")
     except FileNotFoundError:
+        log_str = (
+            "The log_file: %s was not found."
+            + "No logs from the sub-updater will be reported."
+        )
         log.error(
-            f"The log_file: {log_file} was not found.\
-            No logs from the sub-updater will be reported."
+            log_str, log_file,
```
Comment (merlin-northern) [inline src/mender/log/log.py:94]: this is not part of the commit message :)
(nitpick mode: on)

Comment (merlin-northern) [reply src/mender/log/log.py:94]: is there a particular reason why we introducing log_str variable here? it is only used one line below.

File: src/mender/mender.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ def run_daemon(args):
     if args.data:
         log.info(f"Data store set to: {args.data}")
         settings.PATHS = settings.Path(data_store=args.data)
-    statemachine.StateMachine().run(force_bootstrap=args.forcebootstrap)
+    statemachine.run(force_bootstrap=args.forcebootstrap)
```
Comment (merlin-northern) [inline src/mender/mender.py:32]: nice catch!
how on earth did that work?

File: src/mender/client/inventory.py (hunk 1 above)
Comment (oleorhagen) [reply src/mender/client/inventory.py:55]: I must have be mistaken then. Thanks! Good catch!

File: src/mender/log/log.py (hunk 2 above)
Comment (oleorhagen) [reply src/mender/log/log.py:94]: Simply to make `pylint` happy. It didn't understand it when it was declared in the log function :thinking:

File: src/mender/mender.py (hunk 3 above)
Comment (oleorhagen) [reply src/mender/mender.py:32]: It simply skipped checking for the wait file :)
So in the happy case (which is all we are testing right now), it handles just fine :)
I discovered it as I was doing the `failure` case integration tests.

File: src/mender/statemachine/statemachine.py [hunk 4]
```diff
@@ -104,10 +104,15 @@ def __init__(self):
...
-        self.context.deployment_log_handler = deployment_log_handler
+        deployment_log_handler = [
+            handler
+            for handler in log.getLogger("").handlers
+            if isinstance(handler, DeploymentLogHandler)
+        ]
+        assert (
+            len(deployment_log_handler) == 1
+        ), "Something is wrong with the setup of the DeploymentLogHandler"
+        self.context.deployment_log_handler = deployment_log_handler[0]
```
Comment (merlin-northern) [inline src/mender/statemachine/statemachine.py:115]: veryh similar to the section above, perhaps store it in a separate method?

**Review by merlin-northern — APPROVED** [review]
lgtm some comments/questions, I know you are going to make good use of them .

File: src/mender/mender.py (hunk 3 above)
Comment (merlin-northern) [reply src/mender/mender.py:32]: aha!

File: src/mender/log/log.py (hunk 2 above)
Comment (oleorhagen) [reply src/mender/log/log.py:94]: Splitting out into a separate commit.

File: src/mender/statemachine/statemachine.py (hunk 4 above)
Comment (oleorhagen) [reply src/mender/statemachine/statemachine.py:115]: Ideally, we should make our own `rootLogger` subclass,  and use that. But right now I don't have the time for it, so I think this will stay... for now :)

File: src/mender/client/inventory.py (hunk 1 above)
Comment (oleorhagen) [reply src/mender/client/inventory.py]: Removed!
